packages/sol-compiler/src/compiler.ts

```diff
@@ -333,6 +332,69 @@ export class Compiler {
         await fsWrapper.writeFileAsync(currentArtifactPath, artifactString);
         logUtils.warn(`${contractName} artifact saved!`);
     }
+    private _getSourcesWithDependencies(
+        contractPath: string,
+    ): { sourceCodes: { [sourceName: string]: string }; sources: { [sourceName: string]: { id: number } } } {
+        const sources = { [contractPath]: { id: 0 } };
```

Comment:
Could you please verify that the IDs generated that way don't break the source maps? Just check if the coverage still works with those artifacts. Source maps rely on ids geenrated by solidity compiler so you can't just make them auto-increment. Or at least if you do - you'll also need to rewrite the source maps with those new IDs